activerecord/lib/active_record/attribute_methods/serialization.rb

```diff
@@ -23,8 +23,10 @@ module ClassMethods
         # serialized object must be of that class on assignment and retrieval.
         # Otherwise SerializationTypeMismatch will be raised.
         #
-        # Empty objects as <tt>{}</tt>, in the case of +Hash+, or <tt>[]</tt>, in the case of
-        # +Array+, will always be persisted as null.
+        # With default YAMLColumn coder, empty objects as <tt>{}</tt>, in the case of +Hash+,
+        # or <tt>[]</tt>, in the case of +Array+, will always be persisted as null.
+        # This is applicable to any class which responds to +#empty?+.
+        # Therefore, database schema must allow null as default.
```

Comment:
This is not a simplification but a breaking change.
We cannot apply a breaking change without deprecation cycle.